src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasSlicer.java

```diff
@@ -1,258 +1,767 @@
+        if (countryCodeTag.isEmpty())
+        {
+            throw new CoreException("Could not find country code for split relation {}",
+                    newRelation.getIdentifier());
+        }
+        final String countryCode = countryCodeTag.get();
+        final Set<String> synetheticIdsAdded = new HashSet<>();
+        for (int i = 0; i < remainder.getNumGeometries(); i++)
+        {
+            final LineString remainderLine = (LineString) remainder.getGeometryN(i);
+            final Map<String, String> newLineTags = new HashMap<>();
+            newLineTags.put(ISOCountryTag.KEY, countryCode);
+            final EntityIdentifierGenerator lineIdentifierGenerator = new EntityIdentifierGenerator();
+            final CompleteLine newLine = new CompleteLine(0L,
+                    JTS_POLYLINE_CONVERTER.backwardConvert(remainderLine), newLineTags,
+                    new HashSet<Long>());
+            newLine.withIdentifier(lineIdentifierGenerator.generateIdentifier(newLine));
+            newLine.withAddedRelationIdentifier(newRelation.getIdentifier());
+            newRelation.withAddedMember(newLine, role);
+            this.stagedLines.put(newLine.getIdentifier(), newLine);
+            synetheticIdsAdded.add(Long.toString(newLine.getIdentifier()));
+        }
+        return synetheticIdsAdded;
     }
 
-    public String getShardOrAtlasName()
+    /**
+     * Grabs the atlas for the initial shard, in its entirety. Then proceeds to expand out to
+     * surrounding shards if there are any edges bleeding over the shard bounds plus
+     * {@link #SHARD_EXPANSION_DISTANCE}. Finally, will return the constructed Atlas.
+     *
+     * @param initialShard
+     *            The initial {@link Shard} being processed
+     * @param sharding
+     *            The {@link Sharding} used to identify which shards to fetch
+     * @param partiallySlicedAtlasFetcher
+     *            The fetcher policy to retrieve an Atlas file for each shard
+     * @return the expanded {@link Atlas}
+     */
+    private Atlas buildExpandedAtlas(final Shard initialShard, final Atlas initialAtlas,
+            final Sharding sharding, final Function<Shard, Optional<Atlas>> atlasFetcher)
     {
-        return this.shardOrAtlasName;
+        final Predicate<AtlasEntity> expandPredicate = entity -> entity.getType()
+                .equals(ItemType.LINE)
+                && entity.relations().stream()
+                        .anyMatch(relation -> this.relationPredicate.test(relation)
+                                && initialAtlas.relation(relation.getIdentifier()) != null);
+        final Time dynamicAtlasTime = Time.now();
+        logger.info(DYNAMIC_ATLAS_CREATION_TASK, initialShard.getName());
+        final DynamicAtlasPolicy policy = new DynamicAtlasPolicy(atlasFetcher, sharding,
+                initialShard, Rectangle.MAXIMUM).withDeferredLoading(true)
+                        .withExtendIndefinitely(true)
```

Comment:
Is it expected here to extend indefinitely? Especially if the bounds are MAXIMUM?

Reply:
yep. worked on this for a while, this is necessary.